### PR TITLE
Adding return types everywhere

### PR DESCRIPTION
**Pull request by Nek-**
This PR is about adding return types everywhere to get definitely rid of deprecation notice.

Since the code is battle-tested tests should highlight any place when it's wrong. The only class that was not tested is now tested!

This stands as a replacement for #109 .

----------------------------

Since we're using PHP 7.4 as the minimum requirement, there's no problem with this PR. However, the `normalize()` method of the normalizer has union return types. Luckily, we do not need it, so I added the return type string anyway.

Regarding the `declare(strict_types=1)`, I think we should open an issue for version 4.x of the bundle, WDYT?

### CODE REVIEW

File: src/Doctrine/DBAL/Types/PhoneNumberType.php [hunk 1]
```diff
@@ -54,7 +50,7 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      *
      * @return mixed
```
Comment (OskarStark) [inline src/Doctrine/DBAL/Types/PhoneNumberType.php:51]: Can be removed

File: src/Serializer/Normalizer/PhoneNumberNormalizer.php [hunk 2]
```diff
@@ -87,10 +87,10 @@ public function supportsNormalization($data, $format = null, array $context = []
      *
      * @return mixed
```
Comment (OskarStark) [inline src/Serializer/Normalizer/PhoneNumberNormalizer.php:88]: Can be removed

File: src/Serializer/Normalizer/PhoneNumberNormalizer.php [hunk 3]
```diff
@@ -67,7 +67,7 @@ public function __construct(PhoneNumberUtil $phoneNumberUtil, $region = PhoneNum
      *
      * @return array|string|int|float|bool|\ArrayObject|null
```
Comment (OskarStark) [inline src/Serializer/Normalizer/PhoneNumberNormalizer.php:68]: Can be removed

File: tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Misd\PhoneNumberBundle\Tests\DependencyInjection\Compiler;
+
+use Misd\PhoneNumberBundle\DependencyInjection\Compiler\FormTwigTemplateCompilerPass;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class FormTwigTemplateCompilerPassTest extends TestCase
```
Comment (OskarStark) [inline tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php:9]:
```suggestion
final class FormTwigTemplateCompilerPassTest extends TestCase
```

File: tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    public function testItDoesNothingIfThePhoneNumberTwigIsAlreadyConfigured()
+    {
+        $subject = new FormTwigTemplateCompilerPass();
+        $container = new ContainerBuilder();
+        $inputParameter = ['@MisdPhoneNumber/Form/phone_number.html.twig'];
+        $container->setParameter('twig.form.resources', $inputParameter);
+
+        $subject->process($container);
+
+        $this->assertEquals(
```
Comment (OskarStark) [inline tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php:30]: Can we use assertSame here?

File: tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        $container = new ContainerBuilder();
+        $inputParameter = [$sfTemplate];
+        $container->setParameter('twig.form.resources', $inputParameter);
+
+        $subject->process($container);
+
+        $this->assertTrue(\in_array($phoneTemplate, $container->getParameter('twig.form.resources'), true));
+    }
+
+    public function themesProvider()
```
Comment (OskarStark) [inline tests/DependencyInjection/Compiler/FormTwigTemplateCompilerPassTest.php:51]:
```suggestion
    public function themesProvider(): iterable
```

**Comment by Nek-** [comment]
@OskarStark thank you for your review. I fixed everything, and I even found one missing type addition! 👍

**Review by maxhelias — APPROVED** [review]

**Comment by GLacourt** [comment]
Yeah sorry was disconnected for a few weeks. It's ok for me thx. The purpose wasn't a BC break just try to keep up good practices and remove deprecation warning. Another way is to tag version of this bundle and base package to stick with new coding standard and php version. I'm fine with what you did and again thank you !

**Pull request merged**